### PR TITLE
NG UPDATE schematics

### PR DESCRIPTION
**Pull request by hansl**

### CODE REVIEW

File: docs/specifications/update.md [hunk 1]
```diff
@@ -0,0 +1,191 @@
+# Update Command
+
+`ng update` is a new command in the CLI to update one or multiple packages, its peer dependencies, and the peer dependencies that depends on it.
+
+If there are inconsistencies, for example if peer dependencies cannot be matches by a simple semver range, the tool will error out (and nothing will be committed on the filesystem).
+
+## Glossary
+
+* **migration**. A schematic that is performed on the
```
Comment (Brocco) [inline docs/specifications/update.md]: Sentence fragment.
If there are no other terms for the glossary consider omitting it.

File: docs/specifications/update.md [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+        showError('Cannot update dependency "%s": "%s" depends on an incompatible peer dependency' % (x, other))
+
+    update_package_json(p, x)
+```
+
+
+
+## Library Developers
+
+Libraries are responsible for defining their own update schematics. The `ng update` tool will update the package.json, and if it detects a special key in package.json of the library, will run the update schematic on it (with version information metadata).
```
Comment (Brocco) [inline docs/specifications/update.md]: "special key" indicates some sort of secret here, either use it here or mention that it is described below.

File: docs/specifications/update.md (hunk 1 above)
Comment (hansl) [reply docs/specifications/update.md]: Removed the whole Glossary.

File: docs/specifications/update.md (hunk 2 above)
Comment (hansl) [reply docs/specifications/update.md]: Clarified.

**Pull request merged**